packages/cyberstorm/src/components/PackageList/PackageList.tsx

```diff
@@ -22,81 +24,133 @@ interface Props {
 
 const PER_PAGE = 20;
 
-/**
- * Fetches packages based on props and shows them as a list.
- *
- * TODO: add loading indicator over the PackageCard section. Doing so
- *       properly might require rethinking the structure of things, i.e.
- *       adding another wrapping component.
- */
-export function PackageList(props: Props) {
+export function PackageList(props: PackageListProps) {
   const { categories, deprecated, listingType, nsfw, searchQuery, section } =
     props;
 
   const [order, setOrder] = useState(PackageOrderOptions.Updated);
   const [page, setPage] = useState(1);
+  const [packageCount, setPackageCount] = useState<number | null>(null);
```

Comment:
This implementation doesn't seem to work if:

1. The search returns no results (`PackageCount` and `Pagination` components are not rendered, but keep showing the skeleton components)
2. The values are returned from the cache of `usePromise`, at least that's what I think causes it. The issue goes something like this:
    1. Load the initial package list, everything works
    2. Toggle a category filter "include" -> "exclude" -> "off"
    3. Packages are shown as expected, but `PackageCount` and `Pagination` are not

(And TBH feels hacky to start with.)